source/slang/slang-ir-ssa.cpp

```diff
@@ -776,6 +776,19 @@ IRInst* readVar(
         return val;
     }
 
+    if( blockInfo->block == var->parent )
+    {
+        // If this is the block that actually *introduces*
+        // the variable, then there is no reason to keep
+        // searching, because its value cannot have been
+        // established in a predecessor block.
+        //
+        auto type = var->getDataType()->getValueType();
+        val = blockInfo->builder.emitUndefined(type);
```

Comment:
How does this ensure the `undefined` is emitted after the type inst, if that type inst is defined in this same block?

Reply:
For this bug, it doesn’t matter if it is actually emitted in the right order, so long as the *block* if is emitted into is valid to use.

I *think* the logic here in the SSA pass sets the insertion point of the builder to be right before the instruction that reads the value, so the ordering would be fine anyway. If that isn’t the case now, we can fix it to be the case if/when it becomes relevant.